spec.bs

```diff
@@ -310,21 +343,189 @@ maps to an internal [=fenced frame config=] [=struct=].
+  :: null, or a struct containing an [=fencedframetype/exfiltration budget metadata=], a
+     [=visibility/visibility to embedder=], and a [=visibility/visibility to content=]
+
+  : <dfn>nested configs</dfn>
+  :: null, or a struct containing a sequence of pairs of [=urn uuids] and [=fenced frame configs=]
+     and a [=visibility/visibility to embedder=]
+
+  : <dfn>partition nonce</dfn>
+  :: a [=partition nonce=]
+
+  : <dfn>embedder shared storage context</dfn>
+  :: null, or an [=fencedframetype/embedder shared storage context=]
+</dl>
+
+<div algorithm="config instantiation">
+  A [=fenced frame config instance=] is initialized from a [=fenced frame config=] as follows:
+
+    : [=fenced frame config instance/mapped url=]
+    :: the [=fenced frame config=]'s [=fenced frame config/mapped url=]
+
+    : [=fenced frame config instance/container size=]
+    :: the [=fenced frame config=]'s [=fenced frame config/container size=]
+
+    : [=fenced frame config instance/content size=]
+    :: the [=fenced frame config=]'s [=fenced frame config/content size=]
+
+    : [=fenced frame config instance/interest group descriptor=]
+    :: the [=fenced frame config=]'s [=fenced frame config/interest group descriptor=]
+
+    : [=fenced frame config instance/effective sandbox flags=]
+    :: the [=fenced frame config=]'s [=fenced frame config/effective sandbox flags=]
+
+    : [=fenced frame config instance/effective permissions=]
+    :: the [=fenced frame config=]'s [=fenced frame config/effective permissions=]
+
+    : [=fenced frame config instance/fenced frame reporter=]
+    :: the [=fenced frame config=]'s TODO
+
+    : [=fenced frame config instance/exfiltration budget metadata=]
+    :: the [=fenced frame config=]'s TODO
+
+    : [=fenced frame config instance/nested configs=]
+    :: If the [=fenced frame config=]'s [=fenced frame config/nested configs=] is null, set to
+       null. Otherwise, set the [=fenced frame config instance/nested configs=] to an
+       [=ordered map=], where for each [=fenced frame config=] in
+       [=fenced frame config/nested configs=], there is an entry where the key is a randomly
+       sampled [=urn uuid=] and the value is the [=fenced frame config=]. Copy the
+       [=visibility/visibility to embedder=] from the [=fenced frame config=]'s
+       [=fenced frame config/nested configs=]'s [=visibility/visibility to embedder=], and the
+       [=visibility/visibility to content=] from the [=fenced frame config=]'s
+       [=fenced frame config/nested configs=]'s [=visibility/visibility to content=].
+
+    : [=fenced frame config instance/partition nonce=]
+    :: a random [=partition nonce=] (TODO)
+
+    : [=fenced frame config instance/embedder shared storage context=]
+    :: the [=fenced frame config=]'s [=fenced frame config/embedder shared storage context=]
+</div>
+
+<div algorithm=redaction>
+  In order to redact a [=fenced frame config=], run these steps:
```

Comment:
I'd make this:

```html
<div algorithm>
  In other to <dfn export>redact a fenced frame config</dfn>, run these steps:
</div>
```

So that it can be referenced as a dfn from other specs.

Reply:
I changed it to have `<dfn export>redact a [=fenced frame config=]</dfn>`, lmk if the link inside the definition is a problem.

Reply:
Changing the algorithm names to match the models you gave.